### ISSUE TITLE
Regression in incomplete_tag

### ISSUE THREAD

**Issue by Keno** [JuliaLang/JuliaSyntax.jl#512]
On JuliaSyntax master:
```
julia> s = "CompletionFoo.test(1, 1, "
"CompletionFoo.test(1, 1, "

julia> Base.incomplete_tag(Meta.parse(s, raise=false, depwarn=false))
:none
```
vs Base:
```
julia> Base.incomplete_tag(Meta.parse(s, raise=false, depwarn=false))
:other
```

I looked into adjusting the tag selector a bit, but couldn't come up with a good version that also distinguished `[x ` (with space), `[x)` and `(` and `( `, so there may need to be some adjustment to the source-location representation itself. @c42f can you take a look?

**Comment by Keno** [comment]
ref https://github.com/JuliaLang/julia/pull/56110

**Comment by c42f** [comment]
It appears this regression happened in #397

**Comment by c42f** [comment]
@Keno I think this is fixed by #518.  I'm not sure I understand all the distinct test cases you were trying to cover - can you have a quick look and see whether I need to add more?  Thanks!
